daemon/control.c

```diff
@@ -160,6 +164,9 @@ control_ctor(control_t *self, const config_t *config)
 
     /* Init D-Bus service */
     self->ctl_service      = service_create(self);
+
+    /* Set correct session user */
+    self->ctl_session_user = session_current_user(control_session(self));
```

Comment:
That service_create() above kicks async actions potentially in multiple threads -> I'd move this current user fetch so that it happens before that - just to make sure that  the correct value is in place when something that might need it is received over dbus.

Reply:
Good observation. I was pondering a bit how to place it. Moved it.